File: src/main/java/me/zeroeightsix/kami/module/modules/player/NoEntityTrace.java
package me.zeroeightsix.kami.module.modules.player;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;

/**
 * Created by 086 on 8/04/2018.
 */
@Module.Info(name = "NoEntityTrace", category = Module.Category.PLAYER, description = "Blocks entities from stopping you from mining")
public class NoEntityTrace extends Module {

    private static NoEntityTrace INSTANCE;
    private Setting<TraceMode> mode = register(Settings.e("Mode", TraceMode.DYNAMIC));

    public NoEntityTrace() {
        NoEntityTrace.INSTANCE = this;
    }

    public static boolean shouldBlock() {
        return INSTANCE.isEnabled() && (INSTANCE.mode.getValue() == TraceMode.STATIC || mc.playerController.isHittingBlock);
    }

    private enum TraceMode {
        STATIC, DYNAMIC
    }
}

File: src/main/java/me/zeroeightsix/kami/module/modules/player/NoEntityTrace.kt
package me.zeroeightsix.kami.module.modules.player

import me.zeroeightsix.kami.module.Category
import me.zeroeightsix.kami.module.Module
import net.minecraft.item.ItemPickaxe

internal object NoEntityTrace : Module(
    name = "NoEntityTrace",
    category = Category.PLAYER,
    description = "Blocks entities from stopping you from mining"
) {
    private val sneakTrigger = setting("SneakTrigger", false)
    private val pickaxeOnly = setting("PickaxeOnly", true)

    fun shouldIgnoreEntity() = isEnabled && (!sneakTrigger.value || mc.player?.isSneaking == true)
        && (!pickaxeOnly.value || mc.player?.heldItemMainhand?.item is ItemPickaxe)
}